x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/components/add_domain/utils.test.ts

```diff
@@ -21,3 +23,66 @@ describe('extractDomainAndEntryPointFromUrl', () => {
     });
   });
 });
+
+describe('getDomainWithProtocol', () => {
+  const { http } = mockHttpValues;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('passes through domain if it starts with https', async () => {
+    const result = await getDomainWithProtocol('https://elastic.co');
+
+    expect(result).toEqual('https://elastic.co');
+    expect(http.post).toHaveBeenCalledTimes(0);
+  });
+
+  it('passes through domain if it starts with http', async () => {
+    const result = await getDomainWithProtocol('http://elastic.co');
+
+    expect(result).toEqual('http://elastic.co');
+    expect(http.post).toHaveBeenCalledTimes(0);
+  });
+
+  it('returns domain with https protocol if back-end valdates https', async () => {
+    http.post.mockReturnValueOnce(Promise.resolve({ valid: true }));
+    const result = await getDomainWithProtocol('elastic.co');
+
+    expect(result).toEqual('https://elastic.co');
+    expect(http.post).toHaveBeenCalledTimes(1);
+    expect(http.post).toHaveBeenCalledWith('/api/app_search/crawler/validate_url', {
+      body: JSON.stringify({ url: 'https://elastic.co', checks: ['tcp', 'url_request'] }),
+    });
+  });
+
+  it('returns domain with http protocol if back-end valdates http', async () => {
+    http.post.mockReturnValueOnce(Promise.resolve({ valid: false }));
+    http.post.mockReturnValueOnce(Promise.resolve({ valid: true }));
```

Comment:
Super optional, but you can chain this I believe if you're interested
```suggestion
    http.post
      .mockReturnValueOnce(Promise.resolve({ valid: false }))
      .mockReturnValueOnce(Promise.resolve({ valid: true }));
```

Although now I'm not sure that looks nicer though so feel free to ignore this totally lol